src/NodaTime/Duration.cs

```diff
@@ -1154,13 +1177,18 @@ void IXmlSerializable.WriteXml([NotNull] XmlWriter writer)
         [Pure]
         private long ToInt64NanosecondsUnchecked() => unchecked(days * NanosecondsPerDay + nanoOfDay);
 
+        // NOTE: Do not use from internal code. Use decimal instead. TODO: Add an attribute, and use a Roslyn analyzer to validate.
```

Comment:
(Rather than add an attribute, might it be simpler to just disallow any `double` -> `decimal` conversions? It sounds like it would never be a good idea.)